src/wp-includes/blocks/post-comments.php

```diff
@@ -66,3 +66,18 @@ function register_block_core_post_comments() {
 	);
 }
 add_action( 'init', 'register_block_core_post_comments' );
+
+/**
+ * Use the button block classes for the form-submit button.
+ *
+ * @param array $fields The default comment form arguments.
+ *
+ * @return array Returns the modified fields.
+ */
+function gutenberg_post_comments_block_form_defaults( $fields ) {
+	$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="%3$s wp-block-button__link" value="%4$s" />';
+	$fields['submit_field']  = '<p class="form-submit wp-block-button">%1$s %2$s</p>';
+
+	return $fields;
+}
+add_filter( 'comment_form_defaults', 'gutenberg_post_comments_block_form_defaults' );
```

Comment:
The `gutenberg_` prefix should be removed or replaced.

Reply:
We'll have to do this in Gutenberg. I'll create an issue. We really ought to add a lint rule for this since it keeps happening...

Reply:
A lint rule would be helpful to catch these before backporting. Thanks @noisysocks!

Reply:
https://github.com/WordPress/gutenberg/pull/37341